File: python/완전탐색/모의고사.py

""" Level 1. 모의고사 """

def solution(answers):
    
    S1 = [1, 2, 3, 4, 5]
    S2 = [2, 1, 2, 3, 2, 4, 2, 5]
    S3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    check = [0, 0, 0]
    
    for i, num in enumerate(answers):
        
        if S1[i % 5] == num:
            check[0] += 1
        if S2[i % 8] == num:
            check[1] += 1
        if S3[i % 10] == num:
            check[2] += 1
    
    maxCheck = max(check)
    
    answer = [i+1 for i in range(len(check)) if check[i] >= maxCheck]
    
    return answer

print(solution([1, 2, 3, 4, 5]))
print(solution([1, 3, 2, 4, 2]))

File: Java/src/완전탐색/모의고사.java
package 완전탐색;

import java.util.*;

public class 모의고사 {
    
    public static void main(String[] args) {

        int[] a1 = {1,2,3,4,5};
        System.out.println(Arrays.toString(solution(a1)));

        int[] a2 = {1,3,2,4,2};
        System.out.println(Arrays.toString(solution(a2)));
    }

    public static int[] solution(int[] answers) {

        int[] s1 = {1, 2, 3, 4, 5};
        int[] s2 = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] s3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};

        int[] total_sum = new int[3];

        for (int i = 0; i < answers.length; i++) {

            if (answers[i] == s1[i%5]) {
                total_sum[0] += 1;
            }

            if (answers[i] == s2[i%8]) {
                total_sum[1] += 1;
            }

            if (answers[i] == s3[i%10]) {
                total_sum[2] += 1;
            }
        }

        int max_sum = Math.max(total_sum[0], Math.max(total_sum[1], total_sum[2]));

        List<Integer> answer = new ArrayList<>();
        for (int i = 0; i < 3; i++) {
            if (total_sum[i] == max_sum) {
                answer.add(i+1);
            }
        }

        return answer.stream()
                    .mapToInt(Integer::intValue)
                    .toArray();
    }
}
